PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecursiveSoftDrop.cxx

```diff
@@ -0,0 +1,297 @@
+#include "AliEmcalParticle.h"
+#include "AliMCEvent.h"
+#include "AliGenPythiaEventHeader.h"
+#include "AliAODMCHeader.h"
+#include "AliMCEvent.h"
+#include "AliAnalysisManager.h"
+#include "AliJetContainer.h"
+#include "AliParticleContainer.h"
+//#include "AliPythiaInfo.h"
+#include "TRandom3.h"
+#include "AliPicoTrack.h"
+#include "AliEmcalJetFinder.h"
+#include "AliAODEvent.h"
+#include "AliAnalysisTaskRecursiveSoftDrop.h"
+
+#include "FJ_includes.h"
+
+//Globals
+using std::cout;
+using std::endl;
+
+ClassImp(AliAnalysisTaskRecursiveSoftDrop)
+
+//________________________________________________________________________
+AliAnalysisTaskRecursiveSoftDrop::AliAnalysisTaskRecursiveSoftDrop() : 
+  AliAnalysisTaskEmcalJet("AliAnalysisTaskRecursiveSoftDrop", kTRUE),
+  fContainer(0),
+  fJetShapeSub(kNoSub),
+  fPtThreshold(-9999.),
+  fCentSelectOn(kTRUE),
+  fCentMin(0),
+  fCentMax(10),
+  fJetRadius(0.4),
+  fhJetPt(0x0),
+  fhJetPhi(0x0),
+  fhJetEta(0x0),
+  fTreeRecursive_Det(0),
+  fTreeRecursive_True(0)
+
+{
+  for(Int_t i=0;i<nBranches;i++){
+    fShapesVar_Det[i]=0;
+    fShapesVar_True[i]=0;
+  }
+  SetMakeGeneralHistograms(kTRUE);
+  DefineOutput(1, TList::Class());
+  DefineOutput(2, TTree::Class());
+  DefineOutput(3, TTree::Class());
+}
+
+//________________________________________________________________________
+AliAnalysisTaskRecursiveSoftDrop::AliAnalysisTaskRecursiveSoftDrop(const char *name) : 
+  AliAnalysisTaskEmcalJet(name, kTRUE),
+  fContainer(0),
+  fJetShapeSub(kNoSub),
+  fPtThreshold(-9999.),
+  fCentSelectOn(kTRUE),
+  fCentMin(0),
+  fCentMax(10),
+  fJetRadius(0.4),
+  fhJetPt(0x0),
+  fhJetPhi(0x0),
+  fhJetEta(0x0),
+  fTreeRecursive_Det(0),
+  fTreeRecursive_True(0)
+{
+  // Standard constructor.
+  for(Int_t i=0;i<nBranches;i++){
+    fShapesVar_Det[i]=0;
+    fShapesVar_True[i]=0;
+  }
+  SetMakeGeneralHistograms(kTRUE);
+  DefineOutput(1, TList::Class());
+  DefineOutput(2, TTree::Class());
+  DefineOutput(3, TTree::Class());
+}
+
+//________________________________________________________________________
+AliAnalysisTaskRecursiveSoftDrop::~AliAnalysisTaskRecursiveSoftDrop()
+{
+  // Destructor.
+}
+
+//________________________________________________________________________
+ void AliAnalysisTaskRecursiveSoftDrop::UserCreateOutputObjects()
+{
+  // Create user output.
+
+  AliAnalysisTaskEmcalJet::UserCreateOutputObjects();
+
+  Bool_t oldStatus = TH1::AddDirectoryStatus();
+  TH1::AddDirectory(kFALSE);
+  TH1::AddDirectory(oldStatus);
+  //create a tree used for the MC data and making a 4D response matrix
+  const char* nameoutput = GetOutputSlot(2)->GetContainer()->GetName();
+  fTreeRecursive_Det = new TTree(nameoutput, nameoutput);
+  TString *fShapesVarNames_Det = new TString [nBranches];
+  const char* nameoutput2 = GetOutputSlot(3)->GetContainer()->GetName();
+  fTreeRecursive_True = new TTree(nameoutput2, nameoutput2);
+  TString *fShapesVarNames_True = new TString [nBranches];
+  
```

Comment:
Here you allocate memory without freeing it. I recommend replacing by
std::vector<TString> fShapeVarNames_Det(nBranches), nShapeVarNames_True(nBranches);